File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head == null)
            return null;
        int i=0;
        ListNode temp = head;
        while(i < n){
            temp = temp.next;
            i++;
        }
        ListNode temp2 = head;
        while(temp != null && temp.next != null){
            temp = temp.next;
            temp2 = temp2.next;
        }
        if(temp2 == head && temp == null){
            return head.next;
        }
        temp2.next = temp2.next.next;
        return head;
        
    }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        temp = head;
        i=0
        while temp:
            i += 1
            temp = temp.next
        times = i-n-1
        if times < 0:
            return head.next
        temp = head
        while times > 0:
            temp = temp.next
            times -= 1
        temp.next = temp.next.next
        return head
        